Library/Homebrew/test/spec_helper.rb

```diff
@@ -67,11 +69,15 @@
 
       files_after_test = Find.find(TEST_TMPDIR).map { |f| f.sub(TEST_TMPDIR, "") }
 
-      diff = Set.new(@__files_before_test).difference(Set.new(files_after_test))
+      diff = Set.new(@__files_before_test) ^ Set.new(files_after_test)
       expect(diff).to be_empty, <<-EOS.undent
         file leak detected:
         #{diff.map { |f| "  #{f}" }.join("\n")}
       EOS
+
+      Homebrew.failed = @__homebrew_failed
     end
   end
 end
+
+RSpec::Matchers.alias_matcher :have_failed, :be_failed
```

Comment:
Since most of these custom matchers are only used in one file, except for this specific one, I'd like to leave them in their respective specs, and after https://github.com/rspec/rspec-expectations/pull/974 is hopefully accepted, make them scoped. Otherwise, I think, it's easy to forget that they exist.